dvc/remote/base.py

```diff
@@ -820,9 +826,102 @@ def cache_exists(self, checksums, jobs=None, name=None):
         Returns:
             A list with checksums that were found in the remote
         """
-        if not self.no_traverse:
-            return list(set(checksums) & set(self.all()))
 
+        if self.no_traverse:
+            return self._cache_exists_no_traverse(checksums, jobs, name)
+
+        # Fetch one parent cache dir for estimating size of entire remote cache
```

Comment:
comment is a bit misleading ... am I right that we actually fetch not a "parent" dir for some remotes?

Reply:
clarified comment in 65d7e22 since we're really fetching everything under prefix and not parent dirs